### PR TITLE
Make PropertySetter a concrete type

### PR DESCRIPTION
**Pull request by pranavkm**
* Use the pattern from PropertyHelpers to set property values
* Tweaks to Blazor WebAssembly tests to allow running tests locally

### CODE REVIEW

File: src/Components/benchmarkapps/Wasm.Performance/Driver/Selenium.cs [hunk 1]
```diff
@@ -16,12 +16,7 @@ class Selenium
         const int SeleniumPort = 4444;
         static bool RunHeadlessBrowser = true;
 
-        static bool PoolForBrowserLogs =
-#if DEBUG
-            true;
-#else
-            false;
-#endif
+        static bool PoolForBrowserLogs = true;
```
Comment (captainsafia) [inline src/Components/benchmarkapps/Wasm.Performance/Driver/Selenium.cs:19]: Is this intentional?

File: src/Components/Web/src/Forms/ValidationMessage.cs [hunk 2]
```diff
@@ -28,7 +28,7 @@ public class ValidationMessage<TValue> : ComponentBase, IDisposable
         /// <summary>
         /// Specifies the field for which validation messages should be displayed.
         /// </summary>
-        [Parameter] public Expression<Func<TValue>>? For { get; set; }
+        [Parameter(Required = true)] public Expression<Func<TValue>> For { get; set; } = default!;
```
Comment (captainsafia) [inline src/Components/Web/src/Forms/ValidationMessage.cs:31]: Should we set `CurrentEditContext` to be a required cascading parameter as well?

**Review by captainsafia — APPROVED** [review]

**Review by javiercn — APPROVED** [review]
Looks great!

Some minor comments, but that's it.

**Comment by pranavkm** [comment]
@SteveSandersonMS I yanked out all the Required parameters support. I would still like to take the change to use a concrete PropertySetter type and the other benchmark infrastructure changes

**Review by SteveSandersonMS — APPROVED** [review]
Cool

File: src/Components/Components/src/Reflection/IPropertySetter.cs [hunk 3]
```diff
@@ -1,12 +1,55 @@
...
+using System.Reflection;
+
 namespace Microsoft.AspNetCore.Components.Reflection
 {
-    internal interface IPropertySetter
+    internal sealed class PropertySetter
     {
-        bool Cascading { get; }
+        private static readonly MethodInfo CallPropertySetterOpenGenericMethod =
+            typeof(PropertySetter).GetTypeInfo().GetDeclaredMethod(nameof(CallPropertySetter))!;
```
Comment (davidfowl) [inline src/Components/Components/src/Reflection/IPropertySetter.cs:12]: Why are we still using GetTypeInfo?

Comment (pranavkm) [reply src/Components/Components/src/Reflection/IPropertySetter.cs:12]: Yeah, probably unnecessary. 
```suggestion
            typeof(PropertySetter).GetMethod(nameof(CallPropertySetter), BindingFlags.NonPublic | BindingFlags.Static)!;
```

**Review by captainsafia — APPROVED** [review]
Changes look legit.

**Pull request merged**